src/Symfony/Component/Form/Tests/Extension/Core/Type/FormTypeTest.php

```diff
@@ -774,6 +775,67 @@ public function testErrorBubblingDoesNotSkipCompoundFieldsWithInheritDataConfigu
         $this->assertSame($error, $form->get('inherit_data_type')->getErrors()[0]);
         $this->assertCount(0, $form->get('inherit_data_type')->get('child')->getErrors());
     }
+
+    public function testFormAttrOnRoot()
+    {
+        $view = $this->factory
+            ->createNamedBuilder('parent', self::TESTED_TYPE, null, [
+                'form_attr' => true,
+            ])
+            ->add('child1', $this->getTestedType())
+            ->add('child2', $this->getTestedType())
+            ->getForm()
+            ->createView();
+        $this->assertArrayNotHasKey('form', $view->vars['attr']);
+        $this->assertEquals($view->vars['id'], $view['child1']->vars['attr']['form']);
+        $this->assertEquals($view->vars['id'], $view['child2']->vars['attr']['form']);
+    }
+
+    public function testFormAttrOnChild()
+    {
+        $view = $this->factory
+            ->createNamedBuilder('parent', self::TESTED_TYPE)
+            ->add('child1', $this->getTestedType(), [
+                'form_attr' => true,
+            ])
+            ->add('child2', $this->getTestedType())
+            ->getForm()
+            ->createView();
+        $this->assertArrayNotHasKey('form', $view->vars['attr']);
+        $this->assertEquals($view->vars['id'], $view['child1']->vars['attr']['form']);
+        $this->assertArrayNotHasKey('form', $view['child2']->vars['attr']);
+    }
+
+    public function testFormAttrAsBoolWithNoId()
+    {
+        $this->expectException(LogicException::class);
+        $this->expectErrorMessage('form_attr');
+        $this->factory
+            ->createNamedBuilder('', self::TESTED_TYPE, null, [
+                'form_attr' => true,
+            ])
+            ->add('child1', $this->getTestedType())
+            ->add('child2', $this->getTestedType())
+            ->getForm()
+            ->createView();
+    }
+
+    public function testFormAttrAsStringWithNoId()
+    {
+        $stringId = 'custom-identifier';
+        $view = $this->factory
+            ->createNamedBuilder('', self::TESTED_TYPE, null, [
+                'form_attr' => $stringId,
+            ])
+            ->add('child1', $this->getTestedType())
+            ->add('child2', $this->getTestedType())
+            ->getForm()
+            ->createView();
+        $this->assertArrayNotHasKey('form', $view->vars['attr']);
+        $this->assertEquals($stringId, $view->vars['id']);
+        $this->assertEquals($view->vars['id'], $view['child1']->vars['attr']['form']);
+        $this->assertEquals($view->vars['id'], $view['child2']->vars['attr']['form']);
```

Comment:
Oh, please use assertSame when possible in the new test cases.

Reply:
Fixed!